mailpoet/lib/Automation/Integrations/MailPoet/Triggers/OrderStatusChangedTrigger.php

```diff
@@ -0,0 +1,138 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Integrations\MailPoet\Triggers;
+
+use MailPoet\Automation\Engine\Data\StepRunArgs;
+use MailPoet\Automation\Engine\Data\StepValidationArgs;
+use MailPoet\Automation\Engine\Data\Subject;
+use MailPoet\Automation\Engine\Hooks;
+use MailPoet\Automation\Engine\Integration\Trigger;
+use MailPoet\Automation\Integrations\MailPoet\Payloads\OrderPayload;
+use MailPoet\Automation\Integrations\MailPoet\Subjects\OrderSubject;
+use MailPoet\Automation\Integrations\MailPoet\Subjects\SubscriberSubject;
+use MailPoet\Entities\SubscriberEntity;
+use MailPoet\Segments;
+use MailPoet\Subscribers\SubscribersRepository;
+use MailPoet\Validator\Builder;
+use MailPoet\Validator\Schema\ObjectSchema;
+use MailPoet\WooCommerce;
+use MailPoet\WP\Functions;
+
+class OrderStatusChangedTrigger implements Trigger {
+
+
+  /** @var Functions */
+  private $wp;
+
+  /** @var SubscribersRepository */
+  private $subscribersRepository;
+
+  /** @var Segments\WooCommerce */
+  private $woocommerce;
+
+  /** @var WooCommerce\Helper */
+  private $woocommerceHelper;
+
+  public function __construct(
+    Functions $wp,
+    SubscribersRepository $subscribersRepository,
+    Segments\WooCommerce $wooCommerce,
+    WooCommerce\Helper $woocommerceHelper
+  ) {
+    $this->wp = $wp;
+    $this->subscribersRepository = $subscribersRepository;
+    $this->woocommerce = $wooCommerce;
+    $this->woocommerceHelper = $woocommerceHelper;
+  }
+
+  public function getKey(): string {
+    return 'mailpoet:order-status-changed';
+  }
+
+  public function getName(): string {
+    return __('Order status changed', 'mailpoet');
+  }
+
+  public function getArgsSchema(): ObjectSchema {
+
+    return Builder::object([
+      'status' => Builder::string()->required()->default('wc-completed'),
+    ]);
+  }
+
+  public function getSubjectKeys(): array {
+    return [
+      SubscriberSubject::KEY,
+      OrderSubject::KEY,
+    ];
+  }
+
+  public function validate(StepValidationArgs $args): void {
+    // TODO: Implement validate() method.
+  }
+
+  public function registerHooks(): void {
+    add_action(
```

Comment:
Probably a WP helper should be used here.

Reply:
https://github.com/mailpoet/mailpoet/blob/MAILPOET-4935-order-trigger/mailpoet/lib/Automation/Integrations/MailPoet/Triggers/OrderStatusChangedTrigger.php#L77